File: Selenium/Activities/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://v1.training-support.net/selenium/ajax")
    print(driver.title)

    driver.find_element(By.CSS_SELECTOR, "button.violet").click()

    wait = WebDriverWait(driver, 10)

    wait.until(EC.text_to_be_present_in_element((By.TAG_NAME, "h1"), "HELLO!"))

    text = driver.find_element(By.XPATH, "//h1")

    print("Text is : ", text.text)

    wait.until(EC.text_to_be_present_in_element((By.TAG_NAME, "h3"), "I'm late!"))

    lateTxt = driver.find_element(By.TAG_NAME, "h3")
    print("late text is : ", lateTxt.text)
    driver.quit()

File: Selenium/Activities/Activity9.java
package activities;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity9 {

	public static void main(String[] args) {
		WebDriver driver = new FirefoxDriver();
		driver.get("https://v1.training-support.net/selenium/ajax");

		System.out.println("The Title of the page is  : " + driver.getTitle());

		WebElement changeButton = driver.findElement(By.xpath("//button[text()='Change Content']"));
		changeButton.click();
		WebElement txt = driver.findElement(By.xpath("//h1"));
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

		wait.until(ExpectedConditions.visibilityOf(txt));

		System.out.println("Text displayed is : " + txt.getText());

		wait.until(ExpectedConditions.textToBePresentInElementLocated(By.tagName("h3"), "I'm late!"));
		WebElement lateText = driver.findElement(By.tagName("h3"));
		System.out.println("Late text is:  " + lateText.getText());
		driver.quit();
	}

}
